### PR TITLE
Flash and device tree cleanups on nordic

### PR DESCRIPTION
**Pull request by galak**
Some cleanups to get nordic device tree flash support in sync with other platforms.  Also we refactored the dts.fixup files for nordic into SoC out of the board (to reduce duplication).

### CODE REVIEW

File: arch/arm/soc/nordic_nrf5/nrf52/dts.fixup [hunk 1]
```diff
@@ -1,7 +1,10 @@
+/* SoC level DTS fixup file */
+
 #define CONFIG_NUM_IRQ_PRIO_BITS	ARM_V7M_NVIC_E000E100_ARM_NUM_IRQ_PRIORITY_BITS
 #define CONFIG_UART_NRF5_IRQ_PRI	NORDIC_NRF_UART_40002000_IRQ_0_PRIORITY
 #define CONFIG_UART_NRF5_BAUD_RATE	NORDIC_NRF_UART_40002000_CURRENT_SPEED
 #define CONFIG_UART_NRF5_NAME		NORDIC_NRF_UART_40002000_LABEL
 
-#define CONFIG_SOC_FLASH_NRF5_DEV_NAME	NRF_NRF52_FLASH_CONTROLLER_4001E000_LABEL
-#define FLASH_DEV_NAME			CONFIG_SOC_FLASH_NRF5_DEV_NAME
+#define FLASH_DEV_NAME			NRF_NRF52_FLASH_CONTROLLER_4001E000_LABEL
```
Comment (carlescufi) [inline arch/arm/soc/nordic_nrf5/nrf52/dts.fixup]: Should we remove it from all  the soc.h files then?

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: arch/arm/soc/nordic_nrf5/nrf52/dts.fixup (hunk 1 above)
Comment (galak) [reply arch/arm/soc/nordic_nrf5/nrf52/dts.fixup]: Remove what from the soc.h files?

Comment (carlescufi) [reply arch/arm/soc/nordic_nrf5/nrf52/dts.fixup]: Ah sorry, we define `FLASH_DRIVER_NAME` to `FLASH_DEV_NAME` there, confusing :|

**Review by carlescufi — APPROVED** [review]

File: arch/arm/soc/nordic_nrf5/nrf52/dts.fixup (hunk 1 above)
Comment (galak) [reply arch/arm/soc/nordic_nrf5/nrf52/dts.fixup]: Yeah, ideally we'd update mcuboot to use the dt now and remove the need for a define.

**Pull request merged**